### PR TITLE
Landing pages template, views, and urls

### PR DESCRIPTION
**Pull request by zarino**

### CODE REVIEW

File: hub/views/landingpages.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    page_title = "Climate data for your area"
+    meta_description = "" # TODO
+    intro_text = "How much support is there for climate action in your local community? How did your MP vote on environmental issues? What other factors affect climate action in your constituency?"
+    search_form_label = "Input your postcode to see all the climate-related data for your area."
+
+
+class LandingPage02View(BaseLandingPageView):
+    page_title = "Public opinion on climate"
+    meta_description = "" # TODO
+    intro_text = "If you're campaigning for climate or enviornment issues, it's useful to know how much support there is for decarbonisation initiatives in your local area. Who backs climate action, and what percentage of the population backs net zero?"
```
Comment (conatus) [inline hub/views/landingpages.py:39]: Typo in environmental. Hello!
```suggestion
    intro_text = "If you're campaigning for climate or enviromental issues, it's useful to know how much support there is for decarbonisation initiatives in your local area. Who backs climate action, and what percentage of the population backs net zero?"
```

**Comment by lucascumsille** [comment]
@zarino I think I finished customising the pages.

- After talking with @MyfanwyNixon the meta description will be the same as the body or intro_text
- Replaced single straight quotes to curly ones
- Replaces dashes for em dashes when necessary.
- Changed the copy for the filters for something more relevant to the add, but let me know if you think there could be more appropriate ones.
- Added a script to handle the different amounts of characters we can have for h1.



Let me know if you want me to change something.

### Local pages
[Desktop Local Pages.zip](https://github.com/mysociety/local-intelligence-hub/files/14050474/Desktop.Local.Pages.zip)

### Mobile
https://github.com/mysociety/local-intelligence-hub/assets/13790153/0ceb4f4f-981c-4602-a41e-1ad979056e44

### National pages
[Desktop National Pages.zip](https://github.com/mysociety/local-intelligence-hub/files/14050473/Desktop.National.Pages.zip)

### Mobile
https://github.com/mysociety/local-intelligence-hub/assets/13790153/f3e52fdf-47e4-4c64-a33f-1e8c0b7cd58b

**Comment by zarino** [comment]
@struan could we get this deployed to the staging site, so @MyfanwyNixon can have a review, before they go live?

**Review by struan — APPROVED** [review]
added some small fixups to deal with formatting and incorrect img path.

For reference, and not point doing this here, but I think a cleaner way to do this, without so much class and url config boiler plate would be something like:

```python

landing_pages = {
    "this_is_a_path": {
        "page_title": "Public opinion on climate",
        "intro_text": "If you’re campaigning for climate or environmental issues, it’s useful to know how much support there is for decarbonisation initiatives in your local area. Who backs climate action, and what percentage of the population backs net zero?",
        "search_form_label": "Input your postcode to see how much support there is for climate action in your local area.",
        "meta_description": "If you’re campaigning for climate or environmental issues, it’s useful to know how much support there is for decarbonisation initiatives in your local area. Who backs climate action, and what percentage of the population backs net zero? Input your postcode to find out.",
        "example_filter_label": "Believe that the government should continue the block on onshore wind development",
        "example_filter_value": "25%",
        "example_shader_label": "Support renewable energy projects in their local area",
    }
}


class LandingPageView(TitleMixin, TemplateView):
    template_name = "hub/landing_page.html"

    # Default values which could be overridden in the extending Views
    example_filter_label = "Great Big Green Week events 2023"
    example_filter_comparator = "is greater than"
    example_filter_value = "1"
    example_shader_label = "MP majority"

    page_title = "Landing Page"

    def get_context_data(self, **kwargs):
        page = kwargs["page"]
        page_details = landing_pages.get(page, None)
        if page_details is None:
            raise Http404("")

        context = super().get_context_data(**kwargs)
        context["title"] = page_details.get("page_title", "")
        self.page_title = context["title"]
        context["meta_description"] = page_details.get("meta_description", "")
        context["example_filter_label"] = page_details.get("example_filter_label", "")
        context["example_filter_comparator"] = page_details.get(
            "example_filter_comparator", ""
        )
        context["example_filter_value"] = page_details.get("example_filter_value", "")
        context["example_shader_label"] = page_details.get("example_shader_label", "")
        context["intro_text"] = page_details.get("intro_text", "")
        context["search_form_label"] = page_details.get("search_form_label", "")

        if len(context["title"]) < 40:
            context["title_size"] = 3
        elif len(context["title"]) < 80:
            context["title_size"] = 4
        else:
            context["title_size"] = 5

        return context

urlpatterns = [
    path("", LandingPageIndexView.as_view(), name="landing_page_index")
    path("<slug:page>/", LandingPageView.as_view(), name="landing_page"),
]

**Pull request merged**